File: CS/Q588216/Models/MyModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MVCxGridViewDataBinding.Models {
    public class MyModel {
        public MyModel()
            : this(0) {
        }

        public MyModel(int index) {
            ModelID = index;
            ModelName = "Name" + index;
            ModelState = index % 2 == 0;
            ModelDate = DateTime.Now;
        }

        public int ModelID { get; set; }
        [Required]
        public string ModelName { get; set; }
        public Boolean? ModelState { get; set; }
        public DateTime ModelDate { get; set; }

        public static List<MyModel> GetModelList() {
            List<MyModel> l = new List<MyModel>();
            for (int i = 0; i < 100; i++) {
                l.Add(new MyModel(i));
            }
            return l;
        }

        public static void UpdateModel(List<MyModel> l, MyModel m) {
            MyModel editedModel = l.Where(x => x.ModelID == m.ModelID).First();
            editedModel.ModelName = m.ModelName;
            editedModel.ModelDate = m.ModelDate;
            editedModel.ModelState = m.ModelState;
        }

        public static void DeleteModel(List<MyModel> l, int mID) {
            l.Remove(l.Where(x => x.ModelID == mID).First());
        }

        public static void InsertModel(List<MyModel> l, MyModel m) {
            m.ModelID = l.Count + 1;
            l.Add(m);
        }
    }
}

File: VB/Models/MyModel.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Web

Namespace MVCxGridViewDataBinding.Models
	Public Class MyModel
		Public Sub New()
			Me.New(0)
		End Sub

		Public Sub New(ByVal index As Integer)
			ModelID = index
			ModelName = "Name" & index
			ModelState = index Mod 2 = 0
			ModelDate = Date.Now
		End Sub

		Public Property ModelID() As Integer
		<Required>
		Public Property ModelName() As String
		Public Property ModelState() As Boolean?
		Public Property ModelDate() As Date

		Public Shared Function GetModelList() As List(Of MyModel)
			Dim l As New List(Of MyModel)()
			For i As Integer = 0 To 99
				l.Add(New MyModel(i))
			Next i
			Return l
		End Function

		Public Shared Sub UpdateModel(ByVal l As List(Of MyModel), ByVal m As MyModel)
			Dim editedModel As MyModel = l.Where(Function(x) x.ModelID = m.ModelID).First()
			editedModel.ModelName = m.ModelName
			editedModel.ModelDate = m.ModelDate
			editedModel.ModelState = m.ModelState
		End Sub

		Public Shared Sub DeleteModel(ByVal l As List(Of MyModel), ByVal mID As Integer)
			l.Remove(l.Where(Function(x) x.ModelID = mID).First())
		End Sub

		Public Shared Sub InsertModel(ByVal l As List(Of MyModel), ByVal m As MyModel)
			m.ModelID = l.Count + 1
			l.Add(m)
		End Sub
	End Class
End Namespace
